src/software/ai/navigator/path_planner/hrvo/simulated_hrvo_test.py

```diff
@@ -0,0 +1,566 @@
+            [],
+            [],
+            10,
+            False,
+        ),
+        # robot moving straight with a moving enemy robot behind it
+        (
+            [tbots.Point(-2.3, 0)],
+            [tbots.Point(2.8, 0)],
+            [],
+            [tbots.Point(-2.5, 0)],
+            [tbots.Point(-2.1, 0)],
+            10,
+            False,
+        ),
+        # robot moving straight with a moving enemy robot to its side
+        (
+            [tbots.Point(-2.5, 0)],
+            [tbots.Point(2.8, 0)],
+            [],
+            [tbots.Point(-1, -0.8)],
+            [tbots.Point(1, 0)],
+            10,
+            False,
+        ),
+        # robot moving straight with a moving enemy robot moving straight towards it
+        (
+            [tbots.Point(-2.5, 0)],
+            [tbots.Point(2.8, 0)],
+            [],
+            [tbots.Point(2.8, 0)],
+            [tbots.Point(2.5, 0)],
+            10,
+            False,
+        ),
+        # robot moving with a stationary friendly robot in front of it
+        (
+            [tbots.Point(-2.5, 0), tbots.Point(2, 0)],
+            [tbots.Point(2.7, 0), tbots.Point(2, 0)],
+            [],
+            [],
+            [],
+            12,
+            False,
+        ),
+        # robot moving with a stationary enemy robot in front of it
+        (
+            [tbots.Point(0.7, 0)],
+            [tbots.Point(2, 0)],
+            [],
+            [tbots.Point(1, 0)],
+            [],
+            12,
+            False,
+        ),
+        # robot moving straight with a moving friendly robot moving straight towards it
+        (
+            [tbots.Point(-2.5, 0), tbots.Point(2.7, 0)],
+            [tbots.Point(2.7, 0), tbots.Point(-2.5, 0)],
+            [],
+            [],
+            [],
+            10,
+            False,
+        ),
+        # robot moving straight with a moving friendly robot moving straight towards it
+        # while both are turning from 0 to 180 degrees
+        (
+            [tbots.Point(-2.5, 0), tbots.Point(2.7, 0)],
+            [tbots.Point(2.7, 0), tbots.Point(-2.5, 0)],
+            [tbots.Angle.fromRadians(math.pi), tbots.Angle.fromRadians(0)],
+            [],
+            [],
+            10,
+            False,
+        ),
+        # robot moving with a 3 enemy robot wall in front of it
+        (
+            [tbots.Point(0, 0)],
+            [tbots.Point(2.7, 0)],
+            [],
+            [tbots.Point(1, 0), tbots.Point(1, 0.3), tbots.Point(1, -0.3),],
+            [],
+            10,
+            False,
+        ),
+        # robot moving with various stationary enemy robots in front of it
+        # walls generated by range(start_x, start_x + step * num_robots, step)
+        (
+            [tbots.Point(2.8, -1)],
+            [tbots.Point(2.8, 1)],
+            [],
+            [tbots.Point(float(x_meters) / 10, 0) for x_meters in range(20, 32, 2)],
+            [],
+            25,
+            True,
+        ),
```

Comment:
the test where there is a wall of enemy robots in the enemy half and the robot has to travel across doesn't reach the destination and still passes. Is this intentional?

Reply:
For some of the tests there's no validation for reaching the destination since they're more just for observation of what happens on edge cases and are not necessirally deterministic well.